### DISCUSSION TITLE
How to allow querying when requesting from a relationship field but disable on direct query

### DISCUSSION THREAD

**Discussion by fatihsolhan** [keystonejs/keystone#7404] [Questions] [answered]
Hi, I am not sure if the title is self-explanatory but here is the situation:

Assume I have a User and Review list.

User has a field: `reviews: relationship({ ref: 'Review.user', many: true })`
Review has a field: `user: relationship({ ref: 'User.reviews', label: 'Author' })`

I don't want to anyone query my all users so I've added an access filter to my User list as:
```js
access: {
  filter: {
    query: permissions.canManageUsers
  }
}
```

What I want to do is even though I disabled anyone can query my users, I want to show `user.name` field when querying the reviews as
```gql
query {
  reviews {
    comment
    status
    user {
      name
    }
  }
}
```
`user` field is not populating as we expect because I restricted querying it.

So, my question is, is there a way to keep restricting the whole `User` list to be queried while I can read `user.name` field from reviews?

**Comment by Marco-Daniel** [comment, marked as answer]
You can create a virtual field and use `const sudoContext = context.sudo();`  inside the resolver to circumvent the access control

See the docs:
[https://keystonejs.com/docs/guides/auth-and-access-control#circumventing-access-control](https://keystonejs.com/docs/guides/auth-and-access-control#circumventing-access-control)

**Comment by selcukok** [comment]
![Screenshot_2022-06-09-23-54-16-19_e307a3f9df9f380ebaf106e1dc980bb6.jpg](https://user-images.githubusercontent.com/106354809/172951880-6e55d341-39ed-4c9c-ac7a-5260d9d99331.jpg)
